### ISSUE TITLE
NPT costs 2x more

### ISSUE THREAD

**Issue by SparkyTruck** [jax-md/jax-md#330]
The NPT inner step solves the atomic force and box force separately. This makes simulation 2x as slow for systems where the energy function is the major overhead (e.g. neural network potentials). I wonder if it can be merged in a single grad call?

**Comment by abhijeetgangan** [comment]
Yes, it is possible with a single grad call. I will keep in mind to update it down the road.

**Comment by abhijeetgangan** [comment]
I will reopen it as I think it is possible to just store the value. Good catch! Will have a PR shortly.

**Comment by abhijeetgangan** [comment]
Should be solved in #407
